### ISSUE TITLE
Algorithmic Accountability - Journalistic investigation of computational power structures

### ISSUE THREAD

**Issue by aisa6148** [aisa6148/NRS#39]
Nicholas Diakopoulos

**Comment by aisa6148** [comment]
5 cases studied and reverse engineered in journalism context
1. Algorithm makes mistakes and has biases
2. Opaque, difficult to scrutinize
3. "How algorithms exert power and are worthy of scrutiny from computational journalists"
4.  (1) a theoretical lens positing various atomic algorithmic decisions which suggest a number of leading questions that can inform the investigation of algorithms and the development of transparency policies for algorithms 
5. (2) an initial assessment and analysis of how algorithmic accountability is being employed in practice, including its various limitations
6.  assess algorithmic power by analyzing the atomic decisions that algorithms make, including prioritization, classification, association, and filtering.
7. Prioritization:
- Prioritization serves to emphasize or bring attention to certain things at the expense of other things; search engine prioritizes and ranks the most relevant search results.
- prioritization algorithms can enable more efficient management of many social services, such as fire-code inspections, policing and recidivism, immigration enforcement, or welfare management where assigning risk and orienting official attention can provide efficiency gains.
- BUTTTTT is that risk being assigned fairly and with freedom from malice, abuse, or discrimination?
- Sometimes these criteria are not public, making it difficult to understand the weight of different factors contributing to the ranking.
8. Classification:
- categorizing a particular entity as a constituent of a given class by looking at any number of that entity’s features.
- algorithm that makes classification decisions that feed into filtering decisions
- In addition to the potential for uncertainty and mistakes, classification algorithms can also have biases
- Tuning can privilege different stakeholders in a decision, implying an essential value judgment by the designer of such an algorithm in terms of how false positive and false negative errors are balanced 
9. Association
- Association decisions mark relationships between entities and draw their power through both semantics and connotative ability.
- Association algorithms are also built on criteria that define the association, includ- ing the similarity definition of how precisely two things must match to be considered associated with each other
- Association decisions thus also suffer the same kinds of false positive and false negative mistakes as classification decisions.
10. Filtering
-  including or excluding information according to various rules or criteria. 
- Filtering decisions exert their power by either over-emphasizing or censoring certain information. The notion of a “filter bubble” is largely predicated on the idea that by only exposing people to information that they already agree with (by overemphasizing it) it amplifies biases and hampers people’s development of diverse and healthy perspectives
11. IMPORTANT: Algorithmic Accountability
-  algorithms exert power though deci- sions they make in prioritizing, classifying, associating, and filtering information. 
- Through these descriptions, it should also be clear that there are a number of human influences embedded into algorithms, such as criteria choices, training data, semantics, and interpretation.
- Algorithmic accountability must therefore consider algorithms as objects of human creation and take into account intent, including that of any group or institutional processes that may have influenced their design, as well as the agency of human actors in interpreting the output of algorithms in the course of making higher- level decisions.
12. Transparency
- Transparency has recently gained purchase among journalists seeking to build public trust, becoming a new pillar of journalism ethics 
- Government sometimes impose policies to compel disclosure
- algorithm operators may have ulterior goals which conflict with a desire for transparency.
- Oftentimes corporations limit how transparent they are, since exposing too many details (trade secrets) of their proprietary systems may undermine their competitive advantage, hurt their reputation and ability to do business, or leave the system open to gaming and manipulation.
- Transparency is far from a complete solution to balancing algorithmic power. 
- Corporations or governments are not legally or otherwise incentivized to disclose information about their algorithms

The overall goal of this study was to understand the opportunities and limitations of a reverse engineering approach to investigating algorithms. 

In-depth interviews were conducted with four journalists who had worked on or edited stories involving the reverse engineering of algorithms in a news context, including Michael Keller (The Daily Beast), Scott Klein (ProPublica), Jeremy Singer-Vine (Wall Street Journal), and Rob Barry (Wall Street Journal).

Semi structured interviews, analyzed qualitatively

Case studies in reverse engineering
1. Autocompletions on Google and Bing -- "We exclude a narrow class of search queries related to pornography, violence, hate speech, and copyright infringement.” Result: Obvious word were blocked. Most words related to sex and violence weren't
2. Autocorrections on the iPhone. -- The Daily Beast dove into the iPhone spelling correction feature to see which words, like “abortion” or “rape,” the phone would not correct if they were typed incorrectly . Michael’s first attempt to sample this phenomenon was an API on the iPhone, which he used to identify words from a large dictionary that were not getting corrected, essentially pruning down the space of inputs to see what the algorithm “paid attention” to. He noticed that some of the words the API did not correct were getting corrected when they **were** typed directly on an iPhone
3. Targeting political emails -- Propublica sent a survey to users to find out their donations, volunteer history and where they lived to see what parameters were considered to target people with emails. They thought it was the age, they were wrong, it was the donation history
4. Price discrimination in online commerce - different prices to different people. that were adjusting prices dynamically based on different factors like user geography, browser history, or mobile-browser use. In the case of Staples, it found that the input most strongly correlated to price was the distance to a rival’s store, explaining about 90 percent of the pricing pattern.  different computers and browsers in different geographies. built using cookies to see how those user profiles might impact the prices recorded.
One of the issues is that sites like Staples might be using A/B testing to assess different tweaks to their interface. In other words, they are already running experiments on their sites, and to a reverse engineer it might look like noise, or just confusing irregularities. 
Algorithms may be unstable and change over time, or have randomness built in to them, which makes understanding patterns in their input–output relationship much more challenging. Other tactics such as parallelization or analysis of temporal drift may be necessary in order to control for a highly dynamic algorithm.
6. Executive stock trading plans. - sometimes use preset trading plans to avoid accusations of insider trading. The algorithmic plans can be triggered by any number of different parameters, like specific dates, stock prices, or announcements from competitors. The only catch is that the plans cannot be based on inside information.  The Wall Street Journal collected millions of these forms in an attempt to use reverse engineering to see if any of the plans were "opportunistic”. But they couldnt find it because it was hard to tell what inputs the algorithm had
7. Sometimes inputs are hidden or partially hidden, only outputs of the black boxes are available. Very difficult to reverse engineer
8.  “You need to ID algorithms that are very much non-hypothetical and direct in their impact,”
9. Ask questions like: What are the consequences and impact of that algorithm for the public, how significant are those consequences, and how many people might be affected by or perceive an effect by the algorithm? Does the algorithm have the potential for discrimination? Do errors from the algorithm create risks that negatively impact the public or individuals? While newsworthiness criteria for algorithms are still not well-defined, these are but a few examples of questions that might lead to newsworthy investigations of algorithms.

Theory, methodology 
1. Theory - are inputs reverse engineer-able?
2. Identification - what are the consequences?
3. Sampling 
4. Finding the story
5. Reporting

Challenges
1. Teaming computationally literate reporters with tech-savvy computer scientists 
2. It’s a lot of testing or trial and error, and it’s hard to teach in any uniform way
3. More work is also needed to explore the legal ramifications of algorithmic accountability through reverse engineering by journalists.
4. possible ramifications of publishing details of how certain algorithms work. Would publishing such information negatively affect any individuals?

news they report, whether that be in finding new stories in massive datasets or presenting stories interactively, journalistic standards for transparency of algorithms will need to be developed

Several informational dimensions that might be disclosed in a standard transparency policy for algorithms,
 (1) the criteria used to prioritize, rank, emphasize, or editorialize things in the algorithm, including their definitions, operationalizations, and possibly even alternatives; 
(2) what data act as inputs to the algorithm—what it “pays attention” to, and what other parameters are used to initiate the algorithm;
(3) the accuracy including the false positive and false negative rate of errors made in classification (with respect to some agreed-upon ground truth), including the rationale for how the balance point is set between those errors; 
(4) descriptions of training data and its potential bias, including the evolution and dynamics of the algorithm as it learns from data; 
(5) the definitions, operationalizations, or thresholds used by similarity or classification algorithms. The legal and ethical perspectives alluded to above provide an overarching context for how these dimensions might variably be implemented or meet with resistance.
